parquet-column/src/test/java/org/apache/parquet/schema/TestTypeBuilders.java

```diff
@@ -1414,6 +1415,26 @@ public void testTimestampLogicalTypeWithUTCParameter() {
     Assert.assertEquals(nonUtcMicrosExpected, nonUtcMicrosActual);
   }
 
+  @Test
+  public void testVariantLogicalType() {
```

Comment:
Can you test a shredded example as well?

Reply:
I added a simple test below. Initially I added with shredded object and array but for the test here, I think there is no need to have such complexity so I just add the simple test instead.